selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java

```diff
@@ -200,17 +705,146 @@ public void typeToFiltersInput(String value) {
+        .collect(Collectors.toList());
+  }
+
+  public void waitFiltersInputFieldIsEmpty() {
+    seleniumWebDriverHelper.waitInvisibility(By.xpath("//div[@class='md-chip-content']"));
+  }
+
+  public void waitFiltersInputFieldTags(List<String> expectedTags) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>) driver -> getFiltersInputFieldTags().equals(expectedTags));
+  }
+
+  public void deleteTagByRemoveButton(String tagName) {
+    seleniumWebDriverHelper.moveCursorToAndClick(
+        By.xpath(
+            format(
+                "//div[text()='%s']/parent::md-chip-template/div[@class='stack-library-filter-tag-btn']/i",
+                tagName)));
+  }
+
+  public void waitTextContainsInFiltersInput(String expectedText) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> getTextFromFiltersInput().contains(expectedText));
+  }
+
+  public void waitTextEqualsInFiltersInput(String expectedText) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>) driver -> getTextFromFiltersInput().equals(expectedText));
+  }
+
+  public List<String> getFiltersSuggestionsNames() {
+    return seleniumWebDriverHelper
+        .waitVisibilityOfAllElements(By.xpath("//div[@name='suggestionText']"))
+        .stream()
+        .map(webElement -> webElement.getText())
+        .collect(toList());
+  }
+
+  public List<WebElement> getFiltersSuggestions() {
+    return seleniumWebDriverHelper.waitVisibilityOfAllElements(By.xpath("//md-chip"));
+  }
+
+  public void waitFiltersSuggestionsNames(List<String> expectedSuggestions) {
+    expectedSuggestions.forEach(
+        item -> {
+          webDriverWaitFactory
+              .get()
+              .until(
+                  (ExpectedCondition<Boolean>)
+                      driver -> getFiltersSuggestionsNames().contains(item));
+        });
+  }
+
+  public WebElement getSelectedSuggestion() {
+    return getFiltersSuggestions()
+        .stream()
+        .filter(
+            webElement ->
+                webElement
+                    .getAttribute("class")
+                    .contains("stack-library-filter-suggestion-selected"))
+        .collect(toList())
+        .get(0);
+  }
+
+  public void waitSelectedFiltersSuggestion(String suggestionName) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> getSelectedFiltersSuggestionName().equals(suggestionName));
+  }
+
+  public String getSelectedFiltersSuggestionName() {
+    return seleniumWebDriverHelper.waitVisibilityAndGetText(getSelectedSuggestion());
+  }
+
+  public WebElement getFiltersSuggestionByName(String suggestionName) {
+    return seleniumWebDriverHelper.waitVisibility(
+        By.xpath(format("//div[@name='suggestionText' and text()='%s']", suggestionName)));
```

Comment:
`"//div[@name='suggestionText' and text()='%s']"` looks better as a constant among locators, IMHO